unidist/core/backends/mpi/core/serialization.py

```diff
@@ -53,9 +63,14 @@ def is_pickle5_serializable(data):
     bool
         ``True`` if the data should be serialized with pickle using protocol 5 (out-of-band data).
     """
-    return hasattr(data, "__reduce_ex__") and inspect.ismethod(
-        getattr(data, "__reduce_ex__")
-    )
+    is_pandas_instance = False
+    is_numpy_instance = False
+    for mod_name in available_modules:
+        if mod_name == "pandas":
+            is_pandas_instance = isinstance(data, available_modules[mod_name].DataFrame)
+        elif mod_name == "numpy":
+            is_numpy_instance = isinstance(data, available_modules[mod_name].ndarray)
+    return is_pandas_instance or is_numpy_instance
```

Comment:
```suggestion
    for module in available_modules:
        if module.__name__== "pandas":
            is_serializable = isinstance(data, module.DataFrame)
        elif module.__name__ == "numpy":
            is_serializable= isinstance(data, module.ndarray)
    return is_serializable
```

What will we do in a case when we start support other additional complex dtypes? Do you plan to expand this loop/branching?

Reply:
I do not plan to expand compatible modules list in a nearest future as soon as we are mostly working with Modin. We can add additional if/else cases and then think about something better.